core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -684,6 +687,26 @@ Cuda::size_type *cuda_internal_scratch_unified(const Cuda &instance,
 
 namespace Kokkos {
 
+bool Cuda::is_running() const {
+  switch (m_space_instance->m_internal_status) {
+    case Experimental::ExecutionSpaceStatus::complete: return false;
+    case Experimental::ExecutionSpaceStatus::submitted:
+      KOKKOS_IMPL_CUDA_SAFE_CALL(cudaEventRecord(m_space_instance->m_last_event,
+                                                 m_space_instance->m_stream));
+      [[fallthrough]];
+    case Experimental::ExecutionSpaceStatus::running:
+      if (cudaEventQuery(m_space_instance->m_last_event) == cudaSuccess) {
+        m_space_instance->m_internal_status =
+            Experimental::ExecutionSpaceStatus::complete;
+        return false;
+      }
```

Comment:
You need to add sth like
```c++
else
{
m_space_instance->m_internal_status =
            Experimental::ExecutionSpaceStatus::running;
}
```
otherwise you can never get to the `case` directly.

Reply:
Improved.